storio-content-resolver-annotations-processor/src/main/java/com/pushtorefresh/storio/contentresolver/annotations/processor/StorIOContentResolverProcessor.java

```diff
@@ -0,0 +1,331 @@
+                storIOContentResolverColumn
+        );
+    }
+
+    private static void validateAnnotatedClassesAndColumns(@NotNull final Map<TypeElement, StorIOContentResolverTypeMeta> annotatedClasses) {
+        // check that each annotated class has columns with at least one key column
+        for (Map.Entry<TypeElement, StorIOContentResolverTypeMeta> annotatedClass : annotatedClasses.entrySet()) {
+            if (annotatedClass.getValue().columns.isEmpty()) {
+                throw new ProcessingException(annotatedClass.getKey(),
+                        "Class marked with "
+                                + StorIOContentResolverType.class.getSimpleName()
+                                + " annotation should have at least one field marked with "
+                                + StorIOContentResolverColumn.class.getSimpleName()
+                                + " annotation");
+            }
+
+            boolean hasAtLeastOneKeyColumn = false;
+
+            for (final StorIOContentResolverColumnMeta columnMeta : annotatedClass.getValue().columns.values()) {
+                if (columnMeta.storIOContentResolverColumn.key()) {
+                    hasAtLeastOneKeyColumn = true;
+                    break;
+                }
+            }
+
+            if (!hasAtLeastOneKeyColumn) {
+                throw new ProcessingException(annotatedClass.getKey(),
+                        "Class marked with "
+                                + StorIOContentResolverType.class.getSimpleName()
+                                + " annotation should have at least one KEY field marked with "
+                                + StorIOContentResolverColumn.class.getSimpleName() + " annotation");
+            }
+        }
+    }
+    //endregion
+
+    //region Processing of annotated fields
+
+    @Override
+    public synchronized void init(ProcessingEnvironment processingEnv) {
+        super.init(processingEnv);
+        filer = processingEnv.getFiler();
+        elementUtils = processingEnv.getElementUtils(); // why class name is "Elements" but method "getElementUtils()", OKAY..
+        messager = processingEnv.getMessager();
+    }
+
+    @Override
+    public Set<String> getSupportedAnnotationTypes() {
+        final Set<String> supportedAnnotations = new HashSet<String>(2);
+
+        supportedAnnotations.add(StorIOContentResolverType.class.getCanonicalName());
+        supportedAnnotations.add(StorIOContentResolverColumn.class.getCanonicalName());
+
+        return supportedAnnotations;
+    }
+
+    @Override
+    public SourceVersion getSupportedSourceVersion() {
+        return SourceVersion.latestSupported();
+    }
+
+    //endregion
+
+    /**
+     * For those who don't familiar with Annotation Processing API — this is the main method of Annotation Processor lifecycle
+     * <p>
+     * It will be after Java Compiler will find lang elements annotated with annotations from {@link #getSupportedAnnotationTypes()}
```

Comment:
It will be `called` after Java Compiler (looks like I made same mistake in javadoc of sqlite-ap) :)